code/ship/ship.cpp

```diff
@@ -17811,6 +17810,16 @@ float ArmorType::GetDamage(float damage_applied, int in_damage_type_idx, float d
 				constant_val = 0.0f;
 				curr_arg = constant_val;
 			}
+
+			//face: terrible hack to work consistently with beams and additive damages
+			if (is_beam && adtp->Calculations[i] != AT_TYPE_MULTIPLICATIVE
+				&& adtp->Calculations[i] != AT_TYPE_EXPONENTIAL
+				&& adtp->Calculations[i] != AT_TYPE_EXPONENTIAL_BASE
+				&& adtp->Calculations[i] != AT_TYPE_STORE
+				&& adtp->Calculations[i] != AT_TYPE_LOAD) {
```

Comment:
I think a whitelist of armor types that need the hack makes more sense than a blacklist of the types that don't.